freeipa-client/src/main/java/com/sequenceiq/freeipa/client/operation/AbstractFreeIpaAddOperation.java

```diff
@@ -0,0 +1,54 @@
+package com.sequenceiq.freeipa.client.operation;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.client.FreeIpaClientExceptionUtil;
+
+public abstract class AbstractFreeIpaAddOperation<R> extends AbstractFreeipaOperation<R> {
+
+    private String name;
+
+    private AbstractFreeipaOperation<R> getOperation;
+
+    protected AbstractFreeIpaAddOperation(String name) {
+        this.name = name;
+    }
+
+    protected AbstractFreeIpaAddOperation(String name, AbstractFreeipaOperation<R> getOperation) {
+        this.name = name;
+        this.getOperation = getOperation;
+    }
+
+    protected abstract Logger getLogger();
```

Comment:
instead of implementing `getLogger` everywhere, you should use one LOGGER here and using `getOperationName` in the log to identity which add operation was executed

Reply:
It was my first idea but i think it would make more sense to have loggers for the real implementations so in kibana we will have the exact class and we should not find mapping beween command and classes... Can you accept this argument and keep it as it is?

Reply:
it is fine for me